src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java

```diff
@@ -935,25 +938,31 @@ public void testInspectContainerWithSecurityOpts() throws Exception {
 
   @Test
   public void testContainerWithHostConfig() throws Exception {
```

Comment:
I'd prefer to keep this PR small. So maybe include an assumeTrue for this test for docker API <= 1.15 and rename it to something like `testContainerWithHostConfigOldApi`. Then add another `testContainerWithHostConfig` with your changes above?

@rohansingh thoughts?